### DISCUSSION TITLE
I am trying to configure a multi session authentication. I need help because I can't figure it out how to implement it correctly.

### DISCUSSION THREAD

**Discussion by VictorDevCode21** [pocketbase/pocketbase#5464] [Q&A]
I have 2 databases using pocketbase. I am creating an orders panel for an E-commerce website, I've been trying to implement the cookies and authenticacion in both databases at the same time. I've created a login component in Nextjs to authenticate in both databases with the same user and same password. It's weard because when I try to search and view the orders, there's no problem, but when I try to list the items related to the orders in my 2nd database using pocketbase, I think that the cookie of the first database overwrites the 2nd one. I named pb and pb2 for each database. I'll let the code and any related image so you can understand better the problem.  

Auth component: 

![Screenshot 2024-08-31 231724](https://github.com/user-attachments/assets/bdd8b025-44f5-4394-83ed-f1d726b7dfdf)

Context component where I make queries: 

*Here I've created a function to load cookies*
![Screenshot 2024-08-31 231842](https://github.com/user-attachments/assets/04344953-1b90-4cce-91b3-149af652341b)

*Here I try to make queries to retrieve the items related to the orders*
![Screenshot 2024-08-31 231917](https://github.com/user-attachments/assets/c52f5e27-f524-4755-9cf8-3c857da9cff0)

*Here I get the orders and show the orders for the admin user*

![Screenshot 2024-08-31 232056](https://github.com/user-attachments/assets/7ff761f3-b777-4c27-9294-ed956bcae519)

. The problem is that when I try to retrieve the items of the order, pocketbase returns an error, specifically 404 and says auth guest: 

![Screenshot 2024-08-31 232157](https://github.com/user-attachments/assets/65c2d281-2293-413c-a5e2-dc1b5b4c1392)

The weird thing is that when I show the orders to the admin user, the log says 200 and returns the order just fine. But when I try to return the items it's like authentication is gone.

My Api Rule is this: 

@request.auth.id = user.id || @request.auth.role = 'admin'

I can pass trough the authentication to show orders but I can't when I try to return the items related to that order. Is the same API Rule and is the same page, same project, same endpoint, same authentication. Also, when I consoled log the authStore from each database it says that my  pb.authStore.isValid, also my pb2.authStore.isValid, my authStore.model.role is admin in both. Don't know what else I can do.

**Comment by ganigeorgiev** [comment]
_Please don't post code as screenshots - it is hard to read and makes it difficult to search through the code or copy the content in case someone want to assist you and to reproduce your issue._

The easiest way to rule out PocketBase issue is to send the request directly with curl, Postman, Insomnia, etc.

I'm not sure that I understand what is your execution context, how the JS SDK clients were initialized, nor how your authentication works even if I attempt to read the screenshots.

The log clearly shows that the failed request wasn't submitted as authenticated user (as you've noted that's the "Auth: guest" row).

If you are using JS SSR please read https://github.com/pocketbase/pocketbase/discussions/5313 (and moree specially "Problem 1" and "Problem 4") because I suspect that you are either mixing the auth states or that it is caused by the Next.js default cache.

With that said, couple notes:
-  `pb.authStore.isValid` doesn't guarantee that the user is authenticated. It just loosely checks client-side that the store has been populate with a JWT token and its `exp` claim hasn't expired yet, but the actual token verification always happen on the PocketBase server when sending the request.

- I don't know what the `serialize` function does but the code how you set the cookie seems wrong because `pb.authStore.exportToCookie` already returned a serialized cookie string (aka. `pb_auth=...`) and it seems that you are creating a cookie inside another cookie. You can export the cookie with different name by specifying the second argument to `exportToCookie` and `loadFromCookie`:
   ```
   pb.authStore.exportToCookie({ httpOnly: false }, "pb_auth2")
   pb.authStore.loadFromCookie("...", "pb_auth2")
   ```

- If both `pb` and `pb2` SDK clients operates on the client-side then make sure that they use different localStorage keys because otherwise they'll overwrite with each other, ex.:
    ```js
    import PocketBase, { LocalAuthStore } from "pocketbase"

    const pb = new PocketBase("http://127.0.0.1:8090", new LocalAuthStore("pb_auth1"))
    const pb2 = new PocketBase("http://127.0.0.1:8090", new LocalAuthStore("pb_auth2"))
    ```
